client/notifications/index.jsx

```diff
@@ -123,70 +151,95 @@ var Notifications = React.createClass( {
 			event.preventDefault();
 			this.props.checkToggle( null, true );
 		}
-	},
-
-	handleVisibilityChange: function() {
-		this.postMessage( {
-			'action': 'toggleVisibility',
-			'hidden': document.hidden ? true : false
-		} );
-	},
+	};
+
+	handleVisibilityChange = () => this.postMessage( {
+		action: 'toggleVisibility',
+		hidden: !! document.hidden,
+	} );
+
+	/**
+	 * Sends the current oAuth token to the notifications client
+	 *
+	 * Calypso utilizes the browser cookie when making HTTP
+	 * requests and because the notifications client exists in
+	 * the same cookie domain it can make requests without
+	 * needing specific authentication.
+	 *
+	 * However, in the desktop the oAuth token plays a more
+	 * important role in authentication and thus we need to
+	 * pass it into the notifications client so that it can
+	 * make the authenticated calls it needs to.
+	 */
+	postAuth = () => {
+		if ( ! config.isEnabled( 'oauth' ) ) {
+			return;
+		}
 
-	postAuth: function() {
-		if ( config.isEnabled( 'oauth' ) ) {
-			const token = oAuthToken.getToken();
+		const token = oAuthToken.getToken();
 
-			if ( token !== false ) {
-				this.postMessage( {
-					action: 'setAuthToken',
-					token: token
-				} );
-			}
+		if ( token === false ) {
```

Comment:
just because I am trying not to change the behaviors here and it was using strict equality before